NMS-12368: Add an easy to follow step-by-step guide to use Kafka for Minion (#2797)
OpenNMS/opennms: opennms-doc/guide-install/src/asciidoc/text/minion-custom-messaging-system/introduction.adoc

// Allow GitHub image rendering
:imagesdir: ../../images

== Minion with custom messaging system

> A messaging system is used for the communication between _Minions_ and _{opennms-product-name}_ and it comes with _ActiveMQ_ as an embedded solution.

Reviewer: ```suggestion
_Minions_ and _{opennms-product-name}_  communicate via a messaging system.
By default, an embedded _ActiveMQ_ broker is used.
```

> _{opennms-product-name}_ has special needs to a messaging system to run with specific workloads and network topologies a different messaging system than _ActiveMQ_ can be useful.

Reviewer: ```suggestion
_{opennms-product-name}_ is designed to work with different messaging systems and based on the system requirements or workload, an alternative to _ActiveMQ_ can be used.
```

> In general the communication between an _{opennms-product-name}_ instance and _Minions_ is provided by two features:

Reviewer: ```suggestion
In general, the communication between _{opennms-product-name}_ and _Minion_ is provided by two patterns:
```

> * _Remote Producer Calls (RPC)_ are used to issue poller or data collection tasks from an _{opennms-product-name}_ instance to a _Minion_ in a remote location

Reviewer: ```suggestion
* _Remote Producer Calls (RPCs)_ are used to issue specific tasks (such as a request to poll or perform data collection) from an _{opennms-product-name}_ instance to a _Minion_ in a remote location.
** These calls are normally self-contained and include all of the meta-data and information required for them to be performed.
```

> * The _Sink_ feature, which is used to send messages (_Syslog_, _SNMP Traps_ or _Flows_) received from a _Minion_ to an _{opennms-product-name}_ instance

Reviewer: ```suggestion
* The _Sink_ pattern is used to send unsolicited messages (i.e. _Syslog_, _SNMP Traps_ or _Flows_) received from a _Minion_ to an _{opennms-product-name}_ instance
```
